### PR TITLE
feat: Optionally scan the latest release & include Dependabot security alerts

### PR DESCRIPTION
**Pull request by mmcgehee-ccc**

### CODE REVIEW

File: .github/workflows/golang-security-scan.yml [hunk 1]
```diff
@@ -29,20 +35,76 @@ on:
...
+            console.log(`Latest release: ${latestRelease.data.tag_name}`);
+            return latestRelease.data.tag_name;
+      - name: Set checkout refs
+        id: set_checkout_refs
+        run: |
+          if [[ "${{ inputs.scan-latest-release }}" == "true" ]]; then
+            echo "checkout_refs='','${{ steps.get_latest_release.outputs.result }}'" >> $GITHUB_OUTPUT
+          else
+          echo "checkout_refs=''" >> $GITHUB_OUTPUT
+          fi
```
Comment (jwatson-ccc) [inline .github/workflows/golang-security-scan.yml:88]:
```suggestion
          else
            echo "checkout_refs=''" >> $GITHUB_OUTPUT
          fi
```

File: .github/workflows/golang-security-scan.yml [hunk 2]
```diff
@@ -29,20 +35,76 @@ on:
         type: string
         default: "[]" # empty array
         description: Json-formatted array of strings to pass to the docker build command as build args. e.g. '["arg1", "arg2"]'
+      scan-latest-release:
+        description: "Scan the latest release of the repository in addition to the default branch"
+        default: true
+        type: boolean
```
Comment (jwatson-ccc) [inline .github/workflows/golang-security-scan.yml:41]: It looks like this is being required... do we want to require this? 

I was thinking about a drop-in replacement without causing any BC brakage. If it defaults to true, and we do not need to specify, then upgrades would start scanning the latest release automatically. I feel like that would be a safe default and if we explicity do not want to scan the latest release, then we could disable it.

I am trying to decide what would be best.

Comment (jwatson-ccc) [reply .github/workflows/golang-security-scan.yml:41]: But, I see that the DEPENDABOT_ALERTS_TOKEN will case a BC brakage anyway, so maybe that part isn't a big deal.

Comment (mmcgehee-ccc) [reply .github/workflows/golang-security-scan.yml]: Added 'required: false' per our discussion.

**Review by jwatson-ccc — APPROVED** [review]
LGTM

**Pull request merged**